src/View/BaseViewTrait.php

```diff
@@ -53,13 +53,20 @@ private function getProps(): array
     {
         $props = [];
         $only = $this->getPartialData();
-        $onlyViewVars = ! empty($only) ? $only : array_keys($this->viewVars);
+        $onlyViewVars = !empty($only) ? $only : array_keys($this->viewVars);
         $passedViewVars = $this->viewVars;
 
-        $this->viewVars = [];
+        $nonInertiaProps = $this->getConfig('_nonInertiaProps') ?? [];
+
+        $onlyViewVars = array_diff($onlyViewVars, $nonInertiaProps);
+
+        $this->viewVars = array_intersect_key(
+            $passedViewVars,
+            array_flip($nonInertiaProps)
+        );
```

Comment:
Can you describe what is this piece of code doing? I.e. add comment why it's there and brief meaning of what code is doing.